src/openslide-vendor-mirax.c

```diff
@@ -1539,9 +1539,9 @@ static bool mirax_open(openslide_t *osr, const char *filename,
     goto FAIL;
   }
 
-  slidedat = _openslide_read_key_file(tmp, SLIDEDAT_MAX_SIZE,
-                                      G_KEY_FILE_NONE, err);
-  if (!slidedat) {
+  slidedat = g_key_file_new();
+  if (!_openslide_read_key_file(slidedat, tmp, SLIDEDAT_MAX_SIZE,
+                                G_KEY_FILE_NONE, err)) {
```

Comment:
It looks like you changed `_openslide_read_key_file` to take a `GKeyFile*` as the first argument, but this PR doesn't include the corresponding change to the function or other places (vendor-hamamatsu) where it is called from.

Without seeing the changes to `_openslide_read_key_file` I can't be really sure, but this change looks to be introducing a memory leak of the allocated GKeyFile object in the following error path. And if that cleanup happens in the changed function then there is serious mismatch of responsibilities which could easily lead to unintentional NULL pointer dereference errors when callers forget to check the `_openslide_read_key_file` error code.

Reply:
Also, even if the above problems are fixed, it'd be helpful to know the purpose of the change.

Reply:
It was a mistake, the source was downloaded from the website, not github.